v5.1.1 Runtime release

Gauthamastro opened the pull request:
## Describe your changes
Releases the final runtime and client for v5.1.1

zktony committed:
Added benchmark code

zktony committed:
added verify

zktony committed:
added benchmark code

zktony committed:
Added benchmark code

35359595 committed:
initial benchmarks

35359595 committed:
bench machine weights for existing benchmarks

35359595 committed:
thea-message-handler benchmarks implemented

35359595 committed:
runtime integration

35359595 committed:
unused imports cleanup

35359595 committed:
some more unused imports cleanup

35359595 committed:
actual generated weights and integration

35359595 committed:
thea-message-hadler only in benchmarks runtime int

35359595 committed:
taplo fix

zktony committed:
Fixed benchmarks

zktony committed:
Updated trait

35359595 committed:
Update pallets/thea-message-handler/src/weights.rs

Co-authored-by: serhii-temchenko <107397548+serhii-temchenko@users.noreply.github.com>

Gauthamastro committed:
Add extrinsics to update nonces

35359595 committed:
Merge branch 'Develop' into zk-benchmark-43

35359595 committed:
taplo fix

zktony committed:
Removed todo

35359595 committed:
formatting applied

35359595 committed:
Merge branch 'zk-benchmark-43' of github.com:Polkadex-Substrate/Polkadex into zk-benchmark-43

35359595 committed:
benchmarks warnings fixes

35359595 committed:
actual bench machine weights

serhii-temchenko committed:
Iv thea mh benchmarks (#804)

## Describe your changes
Pallet `thea-message-handler` benchmarks, weights and feature gated
runtime integration.

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] I removed all Clippy and Formatting Warnings. 
- [x] I added required Copyrights.

serhii-temchenko committed:
Update outgoing nonce extrinsics (#805)

Adds two extrinsics to update outgoing nonces

35359595 committed:
Merge branch 'Develop' of github.com:Polkadex-Substrate/Polkadex into bench-machine-weights-5.1

35359595 committed:
post #805 PR extrinsics benchmarks

35359595 committed:
benchmarks build fix

serhii-temchenko committed:
feat(no-ref): Added copyrights

Gauthamastro committed:
Update run-a-validator.md

Update port

Gauthamastro committed:
bench machine weights for existing benchmarks (#806)

## Describe your changes
Benchmarks generated for all pallets, except those in separate PRs:
#802 
#805 

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] I removed all Clippy and Formatting Warnings. 
- [x] I added required Copyrights.

zktony committed:
Resolved conflicts

Gauthamastro committed:
Add log

Gauthamastro committed:
Add log

Gauthamastro committed:
Fix snapshot

Gauthamastro committed:
Trace log

35359595 committed:
Fix Snapshot Generation Condition (#809)

## Describe your changes
A bug was introduced in v0.9.43 upgrade that failed snapshot generation
check, this issue fixes that.

Gauthamastro committed:
Added thea-executor benchmark (#802)

## Describe your changes
Added thea-executor benchmark.

serhii-temchenko commented on pallets/thea-message-handler/src/benchmarking.rs:
```diff
@@ -0,0 +1,105 @@
+	}
+}
+
+//#[cfg(test)]
+//use frame_benchmarking::impl_benchmark_test_suite;
+//
+//#[cfg(test)]
+//impl_benchmark_test_suite!(TheaMH, crate::mock::new_test_ext(), crate::mock::Test);
```
Should this be uncommented?

serhii-temchenko commented on pallets/thea-message-handler/src/lib.rs:
```diff
@@ -27,18 +27,27 @@
+	fn insert_authorities(_b: u32) -> Weight;
+	fn incoming_message() -> Weight;
+	fn update_incoming_nonce(_b: u32) -> Weight;
+}
```
Adding required empty line.
```suggestion
}

```

serhii-temchenko commented on pallets/thea-message-handler/src/lib.rs:
```diff
@@ -232,6 +244,21 @@ pub mod pallet {
+
+		/// A governance endpoint to update outgoing nonces
+		#[pallet::call_index(3)]
+		#[pallet::weight(Weight::default())]
```
Should be added  benchmarks and weights?

Gauthamastro replied:
@zktony missed the benchmark for that extrinsic but it is technically same as update incoming nonce, so use it's weight for now.

serhii-temchenko commented on runtime/src/lib.rs:
```diff
@@ -1352,6 +1353,69 @@ impl thea_executor::Config for Runtime {
+		PDEXMigration: pdex_migration::pallet::{Pallet, Storage, Call, Event<T>, Config<T>} = 29,
+		Democracy: pallet_democracy::{Pallet, Call, Storage, Event<T>, Config<T>} = 30,
+		Preimage: pallet_preimage::{Pallet, Call, Storage, Event<T>} = 31,
+		//RandomnessCollectiveFlip: pallet_randomness_collective_flip::{Pallet, Storage} = 32,
```
Should be removed?
```suggestion
```

Gauthamastro replied:
No, until we finalize this change is okay, let it be there,

serhii-temchenko committed:
feature(no-ref): Added changelog update for trlrase 5.1.1

Gauthamastro committed:
Added changelog update for release 5.1.1 (#814)

## Describe your changes

Added changelog update for the release 5.1.1

## Issue ticket number and link

## Checklist before requesting a review
- [X] I have performed a self-review of my code.
- [X] If it is a core feature, I have added thorough tests.
- [X] I removed all Clippy and Formatting Warnings. 
- [X] I added required Copyrights.

Thing todo is to:
- [ ] define proper release date in a changelog before release

Gauthamastro committed:
Update pallets/thea-message-handler/src/lib.rs

Co-authored-by: serhii-temchenko <107397548+serhii-temchenko@users.noreply.github.com>

Gauthamastro committed:
Resolve comments

Gauthamastro committed:
Merge remote-tracking branch 'origin/Develop' into Develop

Gauthamastro committed:
Update versions

Gauthamastro committed:
Update log targets
